src/dmd/dinterpret.d

```diff
@@ -54,8 +54,24 @@ import dmd.visitor;
  */
 public Expression ctfeInterpret(Expression e)
 {
-    if (e.op == TOK.error)
-        return e;
+    switch (e.op)
```

Comment:
Would have been nice to add a comment explaining this is a fast path for frequent use case. People reading this code in a few months might not see it.